app/src/main/java/com/example/drawingactivity/ColorWheel.kt

```diff
@@ -0,0 +1,60 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.util.AttributeSet
+import android.util.Log
+import android.view.View
+
+@SuppressLint("ClickableViewAccessibility")
+class ColorWheel @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    var changeColor: ((Int) -> Unit)? = null
+
+    var wheelColor = Color.BLACK
+    var index = 0
+
+    private var wheelPaint = Paint().apply {
+        color = wheelColor
+        style = Paint.Style.FILL
+    }
+
+    private fun changeWheelColor(i: Int) {
+        changeColor?.invoke(i)
+        invalidate()
+    }
+
+    override fun onDraw(canvas: Canvas?) {
+        super.onDraw(canvas)
+        canvas?.apply {
+            wheelPaint.color = wheelColor
+            drawCircle(width / 2f, height / 2f, 50f, wheelPaint)
+        }
+    }
+
+    init {
+        setOnClickListener {
+            if (index < 3) {
```

Comment:
The 3 shouldn't be a magic number. It should be a constant, or maybe the `.size` of a list

Reply:
moved to viewModel and now calculates using the size of an array